rust/kernel/str.rs

```diff
@@ -186,11 +192,10 @@ impl Index<ops::RangeFrom<usize>> for CStr {
     type Output = CStr;
 
     #[inline]
-    // Clippy false positive
-    #[allow(clippy::unnecessary_operation)]
     fn index(&self, index: ops::RangeFrom<usize>) -> &Self::Output {
         // Delegate bounds checking to slice.
-        &self.as_bytes()[index.start..];
+        // Assign to _ to mute clippy's unnecessary operation warning.
+        let _ = &self.as_bytes()[index.start..];
```

Comment:
Does this fix an unsafe warning?

Reply:
If it is for an unused variable warning, yeah, it should be in a separate commit and ideally a separate PR.

Reply:
Split into a separate commit. Little reluctant to split into a separate PR since I always feel such a simple change don't warrant its own merge commit ;)

Reply:
Yeah, in the future we will not have merge commits -- i.e. patches will be applied on top of each other in a linear fashion (when I add the sentient module to @ksquirrel :-)